tests/src/JIT/Regression/JitBlue/GitHub_23791/GitHub_23791.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+
+using System;
+using System.Runtime.CompilerServices;
+
+// The jit should null check 'this' in NextElement
+
+unsafe struct GitHub_23791
+{
+    fixed byte A[10];
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    byte NextElement(int i) => A[1+i];
+
+    static int Main() 
+    {
+        int result = -1;
+        GitHub_23791* x = null;
+        bool threw = true;
+
+        try
+        {
+            byte t = x->NextElement(100000);
```

Comment:
May want to try a variant like

```cs
byte t = x[offsetThatMakesitValidAddr]->NextElement();
```

Reply:
You are suggesting I figure out how to make `x[y]` point at the start of an actual `GitHub_23791` instance? Seems tricky to do, as the address of the struct is unlikely to be a multiple of `sizeof(GitHub_23791)`.

Reply:
I was thinking about something like:

```cs
using System;
using System.Runtime.CompilerServices;

// The jit should null check 'this' in NextElement

unsafe struct GitHub_23791
{
    fixed byte A[16];

    [MethodImpl(MethodImplOptions.NoInlining)]
    byte NextElement(int i) => A[1 + i];

    static int Main()
    {
        int result = -1;
        GitHub_23791* x = null;
        bool threw = true;

        GitHub_23791 dummy;

        try
        {
            dummy.A[0] = 42;

            byte t = x[(long)(&dummy) / sizeof(GitHub_23791)].A[0];
            System.Console.WriteLine(t);

            t = x->A[(long)(&dummy)];
            System.Console.WriteLine(t);

            threw = false;
        }
        catch (NullReferenceException)
        {
            result = 100;
        }

        if (!threw)
        {
            Console.WriteLine($"FAIL: did not throw an exception");
        }

        return result;
    }
}
```

Currently it prints

```cs
42
42
```

As I understand you want the second one (or both?) to fail.

Reply:
Yes, it may be tricky.   
It works when size is something like `16`, but I am not sure if it is guaranteed.

Anyways, if it is supposed to NRE, then guessing the address exactly may not matter.

Reply:
With the current fix (for windows x64) this prints 42 and then exits with 100, which is what I'd expect (printing 42 is perhaps a happy accident, we may not get that on other ISAs or OSs).

The first instance should not throw, as the base address `x[...]` is accessible; the second should throw, as the base address x is not accessible.